lib/reline/line_editor.rb

```diff
@@ -275,40 +207,27 @@ def reset_variables(prompt = '', encoding:)
     @first_prompt = true
     @searching_prompt = nil
     @first_char = true
-    @add_newline_to_end_of_buffer = false
-    @just_cursor_moving = nil
-    @cached_prompt_list = nil
-    @prompt_cache_time = nil
     @eof = false
     @continuous_insertion_buffer = String.new(encoding: @encoding)
-    @scroll_partial_screen = nil
-    @prev_mode_string = nil
+    @scroll_partial_screen = 0
     @drop_terminate_spaces = false
     @in_pasting = false
     @auto_indent_proc = nil
     @dialogs = []
-    @previous_rendered_dialog_y = 0
-    @last_key = nil
     @resized = false
+    @cursor_y = 0
+    @cache = {}
+    @rendered_screen_cache = nil
     reset_line
   end
 
   def reset_line
-    @cursor = 0
-    @cursor_max = 0
```

Comment:
Can we remove these `instance_variable_set` ? https://github.com/ruby/reline/blob/master/test/reline/test_string_processing.rb#L35-L36

Reply:
Nice catch! I also removed `@line` set in test_string_processing.rb